i18n/ja/docusaurus-plugin-content-docs/current/guides/integration-dsp-no-sdk.md

```diff
@@ -0,0 +1,135 @@
+
+After decrypting the token into a raw UID2, if the token was generated on the client side, you can use the information in `site_data` to verify that a specific domain or app name is on the list of names allowed for it. For details, see [Verify the Domain or App Name](#verify-the-domain-or-app-name).
+
+## Decrypt UID2 Tokens into Raw UID2s
+
+When you have current keys, you'll be able to decrypt a UID2 token into a raw UID2. You also need to check several conditions to make sure that the token is eligible for use in the bidstream.
+
+You'll need to complete the following steps:
+
+1. [Check the token version](#check-the-token-version).
+
+1. [Decrypt the token](#decrypt-the-token).
+
+1. [Make sure the token lifetime is valid](#make-sure-the-token-lifetime-is-valid).
+
+1. [Verify that the domain or app name is valid](#verify-the-domain-or-app-name).
+
+The UID2 SDK for C# / .NET uses a `DecryptTokenIntoRawUid` function to perform these steps: see [BidstreamClient.cs, line 15](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/BidstreamClient.cs#L15).
+
+### Check the Token Version
+
+There are different UID2 token versions in current use, and later processing steps are a little different depending on whether the token version is v2 or a later version.
+
+You can check the token version using the first few bytes of the token.
+
+To review detailed logic, see [UID2Encryption.cs, lines 36-50](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L36-L50).
+
+### Decrypt the Token
+
+Use the master key and site key to decrypt the token. For a code example, refer to the `Decrypt` function: see [UID2Encryption.cs, line 29](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L29). This function decrypts UID2 tokens into raw UID2s as part of the UID2 SDK for C# / .NET, and includes logic to handle different token versions.
+
+### Make Sure the Token Lifetime Is Valid
+
+A token must be valid and current so that it can be used in the bidstream. You must do two things:
+
+- Make sure that the token hasn't expired.
+- Check that the token lifetime is a valid value.
+
+For an example of code that makes sure the token hasn't expired, see [UID2Encryption.cs, line 196](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L196).
+
+To make sure that the token lifetime has a valid value, check these two conditions:
+
+- The remaining lifetime of the token must not exceed the `max_bidstream_lifetime_seconds` value from the `/v2/key/bidstream` response.
```

Comment:
remaining life time is for v2 token, life time is for all later version token, maybe we can say "remaining lifetime/lifetime" here?

Reply:
I feel like we can just say "lifetime" (and not "remaining lifetime") as long as we define what that means somewhere. It could even link to the section showing how to calculate lifetime.

Reply:
Removed all instances of remaining lifetime. Modified the representation slightly, particularly the calculations. This:

For token v2, the calculation to make sure that the token lifetime is valid for bidstream use is as follows:
```
lifetime = token expiry - current time
skew duration = 0
```
and this:
For all token versions later than v2, the calculation to make sure that the token lifetime is valid for bidstream use is as follows:
```
lifetime = token expiry - token generated
skew duration = token generated - current time
```
See update.